### DISCUSSION TITLE
Multiple guards questions

### DISCUSSION THREAD

**Discussion by wasonCheung** [spatie/laravel-permission#2739] [Q&A]
Hi, Can you tell me why I can't assign a role for user? I have two guards that use the same model user. If you know the answer, please let me know. Thank you.

<img width="575" alt="1729249357880" src="https://github.com/user-attachments/assets/5c11a0e6-0a91-4ac3-ad90-56cb7f4749ed">

<img width="483" alt="1729249338903" src="https://github.com/user-attachments/assets/8c60a145-2d50-4f6e-8804-859c0b1858ee">

<img width="706" alt="1729249391061" src="https://github.com/user-attachments/assets/192632f2-52df-4c6e-b0ff-c71556271e0b">

**Comment by wasonCheung** [comment]
I've configured the guardname
<img width="443" alt="1729249479839" src="https://github.com/user-attachments/assets/f985b07e-2947-4efb-8a98-4042a6da6ae6">

**Comment by drbyte** [comment]
Your error message `There is no role named 'super admin' for guard 'portal'` indicates that you're trying to assign a user to a role by that name, specific to that guard. 
And thus the error message is correct, since your database shows only a role for guard `'admin'`.
